cluster/store/store.go

```diff
@@ -918,19 +913,22 @@ func (st *Store) reloadDBFromSnapshot() bool {
 		}
 		return false
 	}
+	st.reloadDBFromSchema()
+	return true
+}
 
+func (st *Store) reloadDBFromSchema() {
```

Comment:
Should we instead have a lock on this function to ensure we don't reload in parallel twice ?

Reply:
yeah, had the same thought but i think `classes := st.db.Schema.MetaClasses()` this will do because it has lock already

Reply:
but also i see the benefits of setting the `lastAppliedIndexOnStart` to 0 , to stop reloading. gonna do that change

Reply:
I think there's a chance that we might be reloading (imagine a large schema so it takes a while) and we trigger a backup restore/snapshot reload and we will change the schema and then start reloading in parallel with the first one.

The first one might start failing I guess ? or the DB might get confused if we're updating. I just want to avoid a different edge case on that problem 😅

Reply:
i agree